python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -1038,6 +1084,24 @@ def do_evaluate(
                 )
                 arg = evaluated.obj_scalar if evaluated.is_scalar else evaluated.obj
             return Column(plc.replace.replace_nulls(column.obj, arg))
+        elif self.name == "pow":
+            column = self.children[0].evaluate(df, context=context, mapping=mapping)
+            if isinstance(self.children[1], Literal):
+                arg = plc.interop.from_arrow(self.children[1].value)
+            else:
+                evaluated = self.children[1].evaluate(
+                    df, context=context, mapping=mapping
+                )
+                arg = evaluated.obj_scalar if evaluated.is_scalar else evaluated.obj
+            return Column(
+                plc.binaryop.binary_operation(
+                    column.obj, arg, plc.binaryop.BinaryOperator.POW, self.dtype
+                )
+            )
+        elif self.name in self._MAPPING:
+            column = self.children[0].evaluate(df, context=context, mapping=mapping)
+            casted = plc.unary.cast(column.obj, self.dtype)
```

Comment:
*question*: why is this necessary? Are you trying to obtain the correct output type? We certainly don't want to cast if the column is _already_ of the correct dtype.

Reply:
Yes, this was to obtain the correct datatype. Usually I assume that casting to the same type is a no-op at some lower level, not sure if that's the case for libcudf.

Reply:
can you show some examples where this cast is necessary?

Reply:
For example, `sin` of an integer column in polars gives float while pylibcudf will return the same dtype as what's input. 

```python
>>> import cudf._lib.pylibcudf as plc
>>> import pyarrow as pa
>>> ary = plc.interop.from_arrow(pa.array([1,2,3]))
>>> result = plc.unary.unary_operation(ary, plc.unary.UnaryOperator.SIN)
>>> plc.interop.to_arrow(result)
<pyarrow.lib.ChunkedArray object at 0x7f4112364770>
[
  [
    0,
    0,
    0
  ]
]

>>> result.type().id()
<type_id.INT64: 4>
>>> import polars as pl
>>> df = pl.DataFrame({'a':[1,2,3]}).lazy()
>>> df.select(pl.col('a').sin())
<LazyFrame at 0x7F3D98CF7490>
>>> df.select(pl.col('a').sin()).collect()
shape: (3, 1)
┌──────────┐
│ a        │
│ ---      │
│ f64      │
╞══════════╡
│ 0.841471 │
│ 0.909297 │
│ 0.14112  │
└──────────┘
```

Reply:
Thanks.